homeassistant/components/notify/slack.py

```diff
@@ -61,9 +68,17 @@ def send_message(self, message="", **kwargs):
         attachments = data.get('attachments') if data else None
 
         try:
-            self.slack.chat.post_message(channel, message,
-                                         as_user=True,
-                                         attachments=attachments,
-                                         link_names=True)
+            if self._username or self._icon:
+                self.slack.chat.post_message(channel, message,
+                                             as_user=False,
+                                             username=self._username,
+                                             icon_emoji=self._icon,
+                                             attachments=attachments,
+                                             link_names=True)
+            else:
+                self.slack.chat.post_message(channel, message,
+                                             as_user=True,
+                                             attachments=attachments,
+                                             link_names=True)
```

Comment:
There seems to be no need for duplicating the `self.slack.chat.post_message()` part. If you follow @MartinHjelmare comment username and icon will be None if they are not present in the configuration.

```python
        if self._username or self._icon:
            self._as_user = False

        try:
            self.slack.chat.post_message(channel, message,
                                         as_user=self._as_user,
                                         username=self._username,
                                         icon_emoji=self._icon,
                                         attachments=attachments,
                                         link_names=True)
...
```